cmd/buildah/containers.go

```diff
@@ -60,38 +95,37 @@ func containersCmd(c *cli.Context) error {
 		return err
 	}
 
-	quiet := c.Bool("quiet")
-	truncate := !c.Bool("notruncate")
-	JSONContainers := []jsonContainer{}
-	jsonOut := c.Bool("json")
+	if c.IsSet("quiet") && c.IsSet("format") {
+		return errors.Errorf("quiet and format are mutually exclusive")
+	}
 
-	list := func(n int, containerID, imageID, image, container string, isBuilder bool) {
-		if jsonOut {
-			JSONContainers = append(JSONContainers, jsonContainer{ID: containerID, Builder: isBuilder, ImageID: imageID, ImageName: image, ContainerName: container})
-			return
-		}
+	opts := containerOptions{
+		all:        c.Bool("all"),
+		format:     c.String("format"),
+		json:       c.Bool("json"),
+		noHeading:  c.Bool("noheading"),
+		noTruncate: c.Bool("notruncate"),
+		quiet:      c.Bool("quiet"),
+	}
 
-		if n == 0 && !c.Bool("noheading") && !quiet {
-			if truncate {
-				fmt.Printf("%-12s  %-8s %-12s %-32s %s\n", "CONTAINER ID", "BUILDER", "IMAGE ID", "IMAGE NAME", "CONTAINER NAME")
-			} else {
-				fmt.Printf("%-64s %-8s %-64s %-32s %s\n", "CONTAINER ID", "BUILDER", "IMAGE ID", "IMAGE NAME", "CONTAINER NAME")
-			}
-		}
-		if quiet {
-			fmt.Printf("%s\n", containerID)
-		} else {
-			isBuilderValue := ""
-			if isBuilder {
-				isBuilderValue = "   *"
-			}
-			if truncate {
-				fmt.Printf("%-12.12s  %-8s %-12.12s %-32s %s\n", containerID, isBuilderValue, imageID, image, container)
-			} else {
-				fmt.Printf("%-64s %-8s %-64s %-32s %s\n", containerID, isBuilderValue, imageID, image, container)
-			}
+	var params *containerFilterParams
+	if c.IsSet("filter") {
+		params, err = parseCtrFilter(c.String("filter"))
+		if err != nil {
+			return errors.Wrapf(err, "error parsing filter")
 		}
+	} else {
```

Comment:
I think this else is unnecessary as `var params *containerFilterParams` is set by default to `nil`.

Reply:
+1, yep we can kill the else.  Just verified on a quick test.

Reply:
fixed. Fixed it in buildah images also.